engine/include/graphics/device.hpp

```diff
@@ -2,22 +2,28 @@
 
 #pragma once
 
+// internal
 #include "core.hpp"
 #include "windowing/window.hpp"
 
+// external
+#include <vulkan/vulkan.hpp>
+
+// std
 #include <vector>
 #include <optional>
 
-#include <vulkan/vulkan.hpp>
-
 namespace gen
 {
-	struct QueueFamilyIndices {
+	struct QueueFamilyIndices
+	{
 		std::optional<u32> graphicsFamily;
+		std::optional<u32> transferFamily;
+		std::optional<u32> presentFamily;
```

Comment:
We want a single family that supports all three operations. Details in other comment.